plugins/aggregators/derivative/README.md

```diff
@@ -0,0 +1,72 @@
+# Derivative Aggregator Plugin
+The Derivative Aggregator Plugin estimates the derivative for all fields of the
+aggregated metrics.
+
+### Time Derivatives
+
+In its default configuration it determines the first and last measurement of
+the period. From these measurements the time difference in seconds is
+calculated. This time difference is than used to divide the difference of each
+field using the following formula:
+```
+              field_last - field_first
+derivative = --------------------------
+                  time_difference
+```
+For each field the derivative is emitted with a naming pattern
+`fieldname_by_seconds`.
+
+### Custom Derivation Variable
+
+The plugin supports to use a field of the aggregated measurements as derivation
+variable in the denominator. This variable is assumed to be a monotonously
+increasing value. In this feature the following formula is used:
+```
+                 field_last - field_first
+derivative = --------------------------------
+              variable_last - variable_first
+```
+For each field the derivative then is emitted with a naming pattern
+`fieldname_by_variablename`.
+
+### Roll-Over to next Period
+
+By default the last measurement is used as first measurement in the next
+aggregation period. This enables a continuous calculation of the derivative. If
+within the next period an earlier timestamp is encountered this measurement will
+replace the roll-over metric. A main benefit of this roll-over is the ability to
+cope with multiple "quiet" periods, where no new measurement is pushed to the
+aggregator. The roll-over will take place at most `max_roll_over`times.
+
+### Configuration
+
+```toml
+[[aggregators.derivative]]
+  ## Specific Derivative Aggregator Arguments:
+  ## Configure a custom derivation variable. Timestamp is used if none is given.
+  # variable = "fieldname_of_derivation_vaiable"
+  ## Infix to separate field name from derivation variable.
+  infix = "_by_"  # default: _by_
```

Comment:
Do we really need this or can this be a `_derivative` suffix. or a `d` prefix or something?

Reply:
The plugin can calculate the derivative of one field with respect to another field, e.g. disk io `read_time` by `read_count`. Using the infix allows the resulting field to be named using both field names by concatenating them with the infix. We can use another naming scheme. Would you prefer a simple suffix instead? It would mean more configuration effort for each use of the plugin, since the user needs to specify the result name more often.

Reply:
To be honest I'm not sure what the best way is. I understood your intention and it is valid. IMO the most common use case is to compute the derivation by a time (even in your case you probably want `d_read_count / dread_time` I guess) and then `dread_count` or `read_count_derivative` is maybe easier to understand for the guy examining the database. But on the other hand this is probably a matter of taste...

So please go ahead with whatever is best in your opinion and mark this as resolved.

Reply:
Keep what you have as I cannot come-up with a better solution :-) and mark this as resolved.